compiler/src/iree/compiler/Codegen/LLVMGPU/Utils/AMDGPUDistributionPatterns.cpp

```diff
@@ -0,0 +1,168 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree-dialects/Dialect/VectorExt/IR/VectorExtOps.h"
+#include "iree/compiler/Codegen/Common/GPU/GPUVectorDistribution.h"
+#include "iree/compiler/Codegen/Common/VectorLayoutAnalysis.h"
+#include "iree/compiler/Codegen/LLVMGPU/Utils/LLVMGPUUtils.h"
+#include "mlir/Dialect/AMDGPU/IR/AMDGPUDialect.h"
+
+namespace mlir::iree_compiler {
+
+using namespace mlir::iree_compiler::IREE::VectorExt;
+using VectorValue = TypedValue<VectorType>;
+
+enum class ContractMatrixType { A, B, C, D };
+enum class ContractType { MM, MMT, MTM, MTMT, UNSUPPORTED };
+
+namespace {
+
+struct DistributeContractions final
+    : OpDistributionPattern<vector::ContractionOp> {
+  using OpDistributionPattern::OpDistributionPattern;
+
+  DistributeContractions(MLIRContext *ctx, MFMAType mfmaType)
+      : OpDistributionPattern(ctx), mfmaType(mfmaType) {}
+
+  SmallVector<int64_t> getIndices(ContractType contractType,
```

Comment:
Can we add some documentation on what this does?

Reply:
Sure I will add some comments.